### ISSUE TITLE
Devise strategy for handling included deposits, exits and slashings

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#275]
## Description

Presently, whenever a `Deposit`, `Exit`, `AttesterSlashing` or `ProposerSlashing` is seen in a block it is deleted from our "inclusion queues" to prevent it being re-included.

This strategy is optimal if there are no forks, however if the `BeaconChain` swaps to another branch where that object has not been included, it will not re-include it again.

## Steps to resolve

Note: this code is currently in #264, do not start issue until it has been merged. Exits queues have not been implemented yet, they will come soon.

Potential solution:

- Keep all objects around until they are included in a finalized block.
- For slashings, only include if the validator is not already penalized.
- For exits, only include if the validator is not already exited.
- For deposits, include based upon

**Comment by paulhauner** [comment]
Closing in favor of #281
